hamlpy/parser/elements.py

```diff
@@ -138,20 +138,24 @@ def __init__(self, tag, _id, classes, attributes, nuke_outer_whitespace, nuke_in
 
         self.classes = class_from_attrs + classes
 
-    def render_attributes(self, attr_wrapper):
+    def render_attributes(self, options):
         def attr_wrap(val):
-            return '%s%s%s' % (attr_wrapper, val, attr_wrapper)
+            return '%s%s%s' % (options.attr_wrapper, val, options.attr_wrapper)
 
         rendered = []
 
         for name, value in self.attributes.items():
-            if name in ('id', 'class'):
-                continue
-
-            if value is None:
-                rendered.append("%s" % name)  # boolean attribute
+            if name in ('id', 'class') or value in (None, False):
+                # this line isn't recorded in coverage because it gets optimized away (http://bugs.python.org/issue2506)
```

Comment:
@nicpottier @ericnewcomer @norkans7 just in case anyone runs into something similar - a line which is definitely hit, but not included in coverage because CPython optimizes it away.